### PR TITLE
OOT: Ageless Hookshot, Climb/Hookshot Anywhere Logic

### PR DESCRIPTION
**Pull request by XenoWars**

### CODE REVIEW

File: packages/core/data/oot/world/jabu_jabu.yml [hunk 1]
```diff
@@ -7,32 +7,32 @@
...
-    "Jabu-Jabu Pot Big Octo Room 2": "event(BIG_OCTO)"
-    "Jabu-Jabu Pot Big Octo Room 3": "event(BIG_OCTO)"
-    "Jabu-Jabu Pot Muscle Block Room 1": "can_boomerang"
-    "Jabu-Jabu Pot Muscle Block Room 2": "can_boomerang"
-    "Jabu-Jabu Pot Muscle Block Room 3": "can_boomerang"
-    "Jabu-Jabu Pot Muscle Block Room 4": "can_boomerang"
-    "Jabu-Jabu Pot Muscle Block Room 5": "can_boomerang"
+    "Jabu-Jabu Pot Big Octo Room 1": "event(BIG_OCTO) || climb_anywhere"
+    "Jabu-Jabu Pot Big Octo Room 2": "event(BIG_OCTO) || climb_anywhere"
+    "Jabu-Jabu Pot Big Octo Room 3": "event(BIG_OCTO) || climb_anywhere"
```
Comment (CelestialKitsune) [inline packages/core/data/oot/world/jabu_jabu.yml:23]: Can you really bypass the big octo fight with climb anywhere? Seems weird to me

File: packages/core/data/oot/world/overworld.yml [hunk 2]
```diff
@@ -869,8 +870,8 @@
     "Kakariko Anju Egg": "is_adult && is_day"
     "Kakariko Anju Cojiro": "event(TALON_AWAKE) && is_day"
     "Kakariko Song Shadow": "is_adult && has(MEDALLION_FOREST) && has(MEDALLION_FIRE) && has(MEDALLION_WATER)"
-    "Kakariko Man on Roof": "can_hookshot || trick(OOT_MAN_ON_ROOF)"
-    "Kakariko GS Roof": "is_adult && gs_night && can_hookshot"
+    "Kakariko Man on Roof": "can_hookshot || trick(OOT_MAN_ON_ROOF) || climb_anywhere"
+    "Kakariko GS Roof": "is_adult && gs_night && (can_hookshot || climb_anywhere)"
```
Comment (CelestialKitsune) [inline packages/core/data/oot/world/overworld.yml:874]: should probably be moved to `Kakariko Rooftop` region

File: packages/core/data/oot/world/shadow_temple.yml [hunk 3]
```diff
@@ -25,17 +25,18 @@
...
+    "Shadow Temple Open": "cond(climb_anywhere, small_keys_shadow(4), small_keys_shadow(1)) && has_explosives"
     "Shadow Temple Scythe Silver Rupees": "true"
   locations:
-    "Shadow Temple Compass": "soul_redead_gibdo && (has_weapon || can_use_sticks || can_use_din)"
+    "Shadow Temple Compass": "soul_redead_gibdo && (has_weapon || can_use_sticks || can_use_din || can_hammer)"
 "Shadow Temple Scythe Silver Rupees":
   dungeon: Shadow
   exits:
     "Shadow Temple Main": "true"
+    "Shadow Temple Boat": "(climb_anywhere && trick(OOT_SHADOW_BOAT_EARLY)) || event(SHADOW_SHORTCUT)"
```
Comment (CelestialKitsune) [inline packages/core/data/oot/world/shadow_temple.yml:36]: how do you get past the block with climb anywhere?

File: packages/core/data/oot/world/shadow_temple.yml [hunk 4]
```diff
@@ -44,75 +45,94 @@
...
-    "Shadow Temple SR Spikes Back Left": "has_lens && can_hookshot"
-    "Shadow Temple SR Spikes Right": "can_hookshot"
+    "Shadow Temple SR Spikes Back Left": "has_lens && (can_hookshot || climb_anywhere)"
+    "Shadow Temple SR Spikes Right": "can_hookshot || climb_anywhere"
     "Shadow Temple SR Spikes Center": "true"
-    "Shadow Temple SR Spikes Front Left": "can_hookshot"
-    "Shadow Temple SR Spikes Midair": "has_lens && can_hookshot"
-    "Shadow Temple Invisible Spike Room": "soul_redead_gibdo && (has_weapon || can_use_sticks || can_use_din)"
+    "Shadow Temple SR Spikes Front Left": "can_hookshot || climb_anywhere"
+    "Shadow Temple SR Spikes Midair": "has_lens && (can_hookshot || climb_anywhere) && (has_weapon || can_use_sticks || can_hammer || has_hover_boots)"
```
Comment (CelestialKitsune) [inline packages/core/data/oot/world/shadow_temple.yml:80]: Can't you just roll jump to it?

File: packages/core/data/oot/world/dodongo_cavern.yml [hunk 5]
```diff
@@ -20,10 +20,10 @@
   dungeon: DC
   exits:
     "Dodongo Cavern Main": "true"
-    "Dodongo Cavern Side Room": "true"
+    "Dodongo Cavern Side Room": "(soul(SOUL_BABY_DODONGO) && (has_weapon || can_use_sticks)) || has_explosives_or_hammer || has_blue_fire_arrows_mudwall"
```
[the remark this answers is not in the document]
Comment (XenoWars) [reply packages/core/data/oot/world/dodongo_cavern.yml:23]: Not true. If your access is Strength, you won't be able to blow it up.

File: packages/core/data/oot/world/fire_temple.yml [hunk 6]
```diff
@@ -36,51 +36,51 @@
     "Fire Temple Boss Key Loop Floor Tiles": "soul(SOUL_FLARE_DANCER)"
     "Fire Temple Boss Key Loop End": "soul(SOUL_FLARE_DANCER)"
   locations:
-    "Fire Temple Boss Key Side Chest": "soul(SOUL_FLARE_DANCER)"
+    "Fire Temple Boss Key Side Chest": "soul(SOUL_FLARE_DANCER) && (is_adult || can_hookshot || climb_anywhere)"
```
[the remark this answers is not in the document]
Comment (XenoWars) [reply packages/core/data/oot/world/fire_temple.yml:39]: That's going towards hammer, not here.

File: packages/core/data/oot/world/fire_temple.yml [hunk 7]
```diff
@@ -36,51 +36,51 @@
...
 "Fire Temple Lava Room":
   dungeon: Fire
   exits:
-    "Fire Temple Maze": "is_adult && small_keys_fire(3) && has_tunic_goron_strict && has(STRENGTH) && (has_ranged_weapon_adult || has_explosives || can_boomerang)"
+    "Fire Temple Maze": "small_keys_fire(3) && has_tunic_goron_strict && (hookshot_anywhere || (((is_adult && has(STRENGTH)) || climb_anywhere) && (has_ranged_weapon || has_explosives)))"
   locations:
-    "Fire Temple Lava Room North Jail Chest": "has_tunic_goron && (is_adult || can_play_time)"
-    "Fire Temple Lava Room South Jail Chest": "is_adult && has_tunic_goron && has_explosives"
-    "Fire Temple GS Lava Side Room": "is_adult && has_tunic_goron && can_play_time"
+    "Fire Temple Lava Room North Jail Chest": "has_tunic_goron"
```
[the remark this answers is not in the document]
Comment (XenoWars) [reply packages/core/data/oot/world/fire_temple.yml:49]: it is not, and i checked.

File: packages/core/data/oot/world/ganon_castle.yml [hunk 8]
```diff
@@ -48,9 +48,9 @@
 "Ganon Castle Forest":
   dungeon: Ganon
   exits:
-    "Ganon Castle Forest 2": "(has_fire_arrows || (can_use_din && has_ranged_weapon_adult))"
+    "Ganon Castle Forest 2": "has_fire_arrows || (can_use_din && (has_ranged_weapon_adult || climb_anywhere))"
```
[the remark this answers is not in the document]
Comment (XenoWars) [reply packages/core/data/oot/world/ganon_castle.yml:51]: can you light the torch from below? the change is so you can climb up to the torch above.

File: packages/core/data/oot/world/ganon_castle.yml [hunk 9]
```diff
@@ -71,13 +71,13 @@
...
-    "Ganon Castle SR Fire Back Right": "has_tunic_goron_strict"
-    "Ganon Castle SR Fire Left": "has_tunic_goron_strict"
-    "Ganon Castle SR Fire Far Right": "has_tunic_goron_strict && can_lift_gold"
-    "Ganon Castle SR Fire Front Right": "has_tunic_goron_strict"
-    "Ganon Castle SR Fire Black Pillar": "has_tunic_goron_strict && can_lift_gold"
+    "Ganon Castle SR Fire Back Right": "has_tunic_goron"
+    "Ganon Castle SR Fire Left": "has_tunic_goron"
+    "Ganon Castle SR Fire Far Right": "has_tunic_goron_strict && (can_lift_gold || climb_anywhere || hookshot_anywhere)"
+    "Ganon Castle SR Fire Front Right": "has_tunic_goron"
+    "Ganon Castle SR Fire Black Pillar": "has_tunic_goron && can_lift_gold"
```
[the remark this answers is not in the document]
Comment (XenoWars) [reply packages/core/data/oot/world/ganon_castle.yml:80]: the checks themselves should be fine. Going to the end room still requires the tunic.

File: packages/core/data/oot/world/ice_cavern.yml [hunk 10]
```diff
@@ -46,22 +42,24 @@
   dungeon: IC
   exits:
     "Ice Cavern Scythe": "true"
-    "Ice Cavern End": "has_blue_fire && has(RUPEE_SILVER_IC_BLOCK, 5)"
+    "Ice Cavern End": "has(RUPEE_SILVER_IC_BLOCK, 5)"
```
[the remark this answers is not in the document]
Comment (XenoWars) [reply packages/core/data/oot/world/ice_cavern.yml:45]: You need to pass red ice to get there from this side regardless, so the blue fire check is redundant.

File: packages/core/data/oot/world/jabu_jabu.yml (hunk 1 above)
Comment (XenoWars) [reply packages/core/data/oot/world/jabu_jabu.yml:23]: yes, you can. either climb up in the second room, or you can climb up to and in in the big octo room itself.

File: packages/core/lib/combo/presets.ts [hunk 11]
```diff
@@ -346,6 +346,15 @@ const PRESET_HELL = makeSettings({
...
     'OOT_FORTRESS_JUMPS',
+	'OOT_COLOSSUS_GS_NO_BEAN',
+	'OOT_ENTER_JABU',
+	'OOT_JABU_BOSS_HIGH_SWITCH',
+	'OOT_WATER_REVERSE_RIVER',
+	'OOT_WATER_PIT_GS_CHU',
+	'OOT_SHADOW_BOAT_EARLY',
+	'OOT_SPIRIT_BOSS_CLIMB_NO_HOOK',
+	'OOT_DC_BOULDER',
+	'OOT_LAKE_SHORTCUT',
```
[the remark this answers is not in the document]
Comment (XenoWars) [reply packages/core/lib/combo/presets.ts:357]: i do use spaces. it's just a matter of pressing enter to move to a new line.

File: packages/core/data/oot/world/overworld.yml [hunk 12]
```diff
@@ -95,14 +95,13 @@
...
-    "Kokiri Gossip Near Lost Woods Grotto": "hidden_grotto_storms"
 "Kokiri Forest Near Deku Tree":
   region: KOKIRI_FOREST
   exits:
-    "Kokiri Forest": "mido_moved || is_adult"
+    "Kokiri Forest": "mido_moved || is_adult || climb_anywhere || hookshot_anywhere"
     "Deku Tree": "is_child || (is_adult && setting(dekuTreeAdult) && mido_moved)"
   events:
-    STICKS: "can_boomerang || (is_child && has_weapon)"
+    STICKS: "is_child && (can_boomerang || has_weapon)"
```
[the remark this answers is not in the document]
Comment (XenoWars) [reply packages/core/data/oot/world/overworld.yml:104]: adult can't get sticks here due to forest temple clear removing all enemies. but yeah, does need deku baba soul.

File: packages/core/data/oot/world/shadow_temple.yml (hunk 3 above)
Comment (XenoWars) [reply packages/core/data/oot/world/shadow_temple.yml:36]: you just climb up the cage. you pop up on top.

File: packages/core/data/oot/world/shadow_temple.yml (hunk 4 above)
Comment (XenoWars) [reply packages/core/data/oot/world/shadow_temple.yml:80]: nope, too far.

File: packages/core/data/oot/world/spirit_temple.yml [hunk 13]
```diff
@@ -17,14 +17,14 @@
...
-    "Spirit Temple Child Climb": "is_child && small_keys_spirit(1)"
+    "Spirit Temple Child Climb": "is_child && cond(climb_anywhere, (small_keys_spirit(1) && has_mirror_shield && has_explosives) || small_keys_spirit(2), small_keys_spirit(1))" #1 is safe if you have the means to lower the light platform, otherwise 2 is safe.
     "Spirit Temple Child Left": "soul_keese && soul_armos && (can_use_sticks || has_explosives_or_hammer || ((can_boomerang || has_nuts) && (has_weapon || can_use_slingshot)))"
     "Spirit Temple Child Right": "soul_keese && soul_armos && (can_use_sticks || has_explosives_or_hammer || ((can_boomerang || has_nuts) && (has_weapon || can_use_slingshot)))"
 "Spirit Temple Child Left":
   dungeon: Spirit
   exits:
     "Spirit Temple Child Entrance": "true"
-    "Spirit Temple Child Back Left": "has_ranged_weapon_child"
+    "Spirit Temple Child Back Left": "has_ranged_weapon_child || hookshot_anywhere || climb_anywhere || (can_hookshot && can_jump_slash) || can_longshot"
```
[the remark this answers is not in the document]
Comment (XenoWars) [reply packages/core/data/oot/world/spirit_temple.yml:27]: You can hook onto the grated floor.

File: packages/core/data/oot/world/shadow_temple.yml (hunk 4 above)
Comment (XenoWars) [reply packages/core/data/oot/world/shadow_temple.yml:80]: I am reminded, though, that hookshot anywhere as adult or longshot anywhere as child should be able to reach it.

**Comment by CelestialKitsune** [comment]
Everything is good on my end. If there's any issue, we'll fix them on the fly.

**Pull request merged**